content/en/docs/installation/authentication.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Authentication"
+weight: 40
+description: >
+  Guide to set up basic authentication with Fission
+---
+
+## Authentication for Fission Functions
+
+When using Fission, if you are using an ingress, you might already have some form of authentication in place for external calls.
+But if you aren't, Fission didn't provide a way to configure authentication for your API calls.
+
+Starting with [v1.16.0](/docs/releases/v1.16.0-rc1/), Fission allows you to have an **authentication mechanism in place for Fission function invocations**.
+
+## Understanding Authentication for Fission Functions
+
+Fission does so by enabling authentication for Fission Router.
+This is an optional feature that can be enabled/disable depending on your requirement.
+
+When enabled, a new endpoint for authentication will be registered in the router.
+All the API calls to Fission functions will now be routed through this endpoint.
+
+Fission will also create a secret named `router` in `fission` namespace with a default `username`, a randomly generated `password` and a `jwtSigninKey`.
+This secret is mounted on a volume on the router pod.
+The user has to first create an `auth token` by providing the `username` and `password`.
+The token generated has to be passed in the authentication header for all subsequent API calls to the function.
+
+## Enabling Authentication
+
+To enable authentication, you need to set the key `authentication.enabled` to `true`.
+This can be found in `charts/fission-all/values.yaml`.
+
+```bash
+--set authentication.enabled=true
+```
+
+You can also set other parameters related to Authentication
+
+```yaml
+  ## authUriPath defines authentication endpoint path via router 
+  ## default '/auth/login'
+  authUriPath:
+
+  ## authUsername is used as a username for authentication
+  ## default 'admin'
+  authUsername: admin
+  
+  ## jwtSigningKey is the signing key used for signing the JWT token
+  jwtSigningKey: serverless
+
+  ## jwtExpiryTime is the JWT expiry timein seconds
+  ## default '120'
+  jwtExpiryTime: 
+  
+  ## jwtIssuer is the issuer of JWT
+  ## default 'fission'
+  jwtIssuer: fission
+```
+
+>> Refer to our [installation](_index.en.md) guide if you are installing Fission for the first time. Or visit our [Fission Upgrade](upgrade) guide if you're upgrading from an older version.
+
```

Comment:
Please add a separate section here for the token generation part rather than keeping it under enabling auth.

```
## Generating auth token
```

Also we should link doc for CLI here 
https://deploy-preview-155--fission-website.netlify.app/docs/reference/fission-cli/fission_token_create/